File: lib_base/src/main/java/com/android/base/app/ui/RefreshLoadMoreView.java
package com.android.base.app.ui;

/**
 * <br/>   对下拉刷新的抽象
 * <br/>   Email: 1169654504@qq.com
 *
 * @author Ztiany
 * @version 1.0
 */
public interface RefreshLoadMoreView {

    void autoRefresh();

    void refreshCompleted();

    void loadMoreCompleted(boolean hasMore);

    void loadMoreFailed();

    void setRefreshHandler(RefreshHandler refreshHandler);

    void setLoadMoreHandler(LoadMoreHandler loadMoreHandler);

    boolean isRefreshing();

    boolean isLoadingMore();

    void setRefreshEnable(boolean enable);

    void setLoadMoreEnable(boolean enable);

    interface RefreshHandler {
        void onRefresh();
    }

    interface LoadMoreHandler {
        void onRefresh();
    }

}

File: lib_base/src/main/java/com/android/base/architecture/ui/RefreshLoadMoreView.kt
package com.android.base.architecture.ui

/**
 * <br></br>   对下拉刷新的抽象
 * <br></br>   Email: 1169654504@qq.com
 *
 * @author Ztiany
 * @version 1.0
 */
interface RefreshLoadMoreView {

    fun autoRefresh()

    fun refreshCompleted()

    fun loadMoreCompleted(hasMore: Boolean)

    fun loadMoreFailed()

    fun setRefreshHandler(refreshHandler: RefreshHandler?)

    fun setLoadMoreHandler(loadMoreHandler: LoadMoreHandler?)

    fun isRefreshing(): Boolean

    fun isLoadingMore(): Boolean

    var isRefreshEnable: Boolean

    var isLoadMoreEnable: Boolean

    interface RefreshHandler {
        fun onRefresh()
    }

    interface LoadMoreHandler {
        fun onLoadMore()
    }

}